### ISSUE TITLE
SIGSEGV when executing read_csv query for struct with empty varchar

### ISSUE THREAD

**Issue by ryderblue** [duckdb/duckdb#9447]
### What happens?

SIGSEGV (or runtime assertion error with debug build) for simple `read_csv()` query for `STRUCT` with empty `VARCHAR` value: `Error: INTERNAL Error: Assertion triggered in file "/Users/ryderblue/dev/cpp/duckdb/src/common/types/string_heap.cpp" on line 24: Utf8Proc::Analyze(data, len) != UnicodeType::INVALID`

### To Reproduce

1. Create the following `test.csv` file:
```
"{'val': }"
```
2. Run query:
```
duckdb -c "SELECT * FROM read_csv('test.csv', columns={'col': 'STRUCT(val VARCHAR)'}) ORDER BY 1;" test
```
```
Segmentation fault: 11
```

### OS:

Ubuntu 20.04 x86_64

### DuckDB Version:

0.9.1

### DuckDB Client:

CLI

### Full Name:

Ryder Rishel

### Affiliation:

Zoox

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by ryderblue** [comment]
This can be closed due to #9448 merging.

**Comment by szarnyasg** [comment]
Thanks for the fix!
